pkg/crypto/tinkcrypto/primitive/composite/ecdhes/subtle/ecdhes_aead_composite_test.go

```diff
@@ -0,0 +1,234 @@
+		mEncHelper)
+	require.NoError(t, err)
+
+	// Encrypt should fail with large AEAD key size value
+	_, err = cEnc.Encrypt(pt, aad)
+	require.Error(t, err)
+
+	mEncHelper.KeySizeValue = 32
+
+	// test with GetAEAD() returning error
+	mEncHelper.AEADErrValue = fmt.Errorf("error from GetAEAD")
+
+	cEnc, err = NewECDHESAEADCompositeEncrypt(recipientsPubKeys, commonpb.EcPointFormat_UNCOMPRESSED.String(),
+		mEncHelper)
+	require.NoError(t, err)
+
+	// Encrypt should fail with large AEAD key size value
+	_, err = cEnc.Encrypt(pt, aad)
+	require.EqualError(t, err, "error from GetAEAD")
+
+	mEncHelper.AEADErrValue = nil
+
+	// create a valid ciphertext to test Decrypt for all recipients
+	cEnc, err = NewECDHESAEADCompositeEncrypt(recipientsPubKeys, commonpb.EcPointFormat_UNCOMPRESSED.String(),
+		mEncHelper)
+	require.NoError(t, err)
+
+	// test with empty plaintext
+	ct, err := cEnc.Encrypt([]byte{}, aad)
+	require.NoError(t, err)
+
+	encData := new(EncryptedData)
+	err = json.Unmarshal(ct, encData)
+
+	require.NoError(t, err)
+	// encrypting empty plaintext should result in empty ciphertext
+	require.Empty(t, encData.Ciphertext)
+	require.Len(t, encData.Tag, subtleaead.AESGCMTagSize)
+	require.Len(t, encData.IV, subtleaead.AESGCMIVSize)
+
+	ct, err = cEnc.Encrypt(pt, aad)
+	require.NoError(t, err)
+
+	for _, privKey := range recipientsPrivKeys {
+		// test with nil recipient private key
+		dEnc, err := NewECDHESAEADCompositeDecrypt(nil, commonpb.EcPointFormat_UNCOMPRESSED.String(), mEncHelper)
+		require.NoError(t, err)
+
+		_, err = dEnc.Decrypt(ct, aad)
+		require.Error(t, err)
+
+		// test with large key size
+		mEncHelper.KeySizeValue = 100
+		dEnc, err = NewECDHESAEADCompositeDecrypt(privKey, commonpb.EcPointFormat_UNCOMPRESSED.String(), mEncHelper)
+		require.NoError(t, err)
+
+		_, err = dEnc.Decrypt(ct, aad)
+		require.Error(t, err)
+
+		mEncHelper.KeySizeValue = 32
+
+		// test with GetAEAD() returning error
+		mEncHelper.AEADErrValue = fmt.Errorf("error from GetAEAD")
+
+		dEnc, err = NewECDHESAEADCompositeDecrypt(privKey, commonpb.EcPointFormat_UNCOMPRESSED.String(), mEncHelper)
+		require.NoError(t, err)
+
+		_, err = dEnc.Decrypt(ct, aad)
+		require.Error(t, err)
+
+		mEncHelper.AEADErrValue = nil
+
+		// create a valid Decrypt message and test against ct
+		dEnc, err = NewECDHESAEADCompositeDecrypt(privKey, commonpb.EcPointFormat_UNCOMPRESSED.String(), mEncHelper)
+		require.NoError(t, err)
+
+		// try decrypting empty ct
+		_, err = dEnc.Decrypt([]byte{}, aad)
+		require.Error(t, err)
+
+		// finally try successful decrypt
+		dpt, err := dEnc.Decrypt(ct, aad)
+		require.NoError(t, err)
+		require.EqualValues(t, pt, dpt)
+	}
+}
+
+func buildRecipientsKeys(t *testing.T, nbOfRecipients int) ([]*hybrid.ECPrivateKey, []*hybrid.ECPublicKey) {
+	t.Helper()
+
+	var (
+		recipientsPrivKeys []*hybrid.ECPrivateKey
+		recipientsPubKeys  []*hybrid.ECPublicKey
+	)
+
+	curvProto := commonpb.EllipticCurveType_NIST_P256
```

Comment:
need an additional test for 25519

Reply:
That's for signing, not encryption. 

The new primitive only deals with encryption (to support JWE serialization). 

For JWS, we can use Tink's Ed25519 signer and verifier primitives directly which are currently supported by localkms and crypto. We have a test that exports and imports Ed25519 keys and do signing/verification [here](https://github.com/hyperledger/aries-framework-go/blob/master/pkg/kms/localkms/pubkey_export_import_test.go#L50).

Reply:
X25519

Reply:
I added a TODO for X25519 key wrapping.